src/Adaptivity/AdaptedTriangulations.jl

```diff
@@ -0,0 +1,316 @@
+  end
+
+  # D) Different background model AND different type of triangulation
+  @notimplemented
+  return nothing
+end
+
+function Geometry.best_target(strian::AdaptedTriangulation,ttrian::Triangulation)
+  @check is_change_possible(strian,ttrian)
+  return strian
+end
+
+function Geometry.best_target(strian::Triangulation,ttrian::AdaptedTriangulation)
+  @check is_change_possible(strian,ttrian)
+  return ttrian
+end
+
+function CellData.change_domain(a::CellField,strian::Triangulation,::ReferenceDomain,ttrian::AdaptedTriangulation,::ReferenceDomain)
+  if strian === ttrian
+    return a
+  end
+  @assert is_change_possible(strian,ttrian)
+
+  if (get_background_model(strian) === get_background_model(ttrian))
+    return change_domain(a,strian,ReferenceDomain(),ttrian.trian,ReferenceDomain())
+  end
+  
+  return change_domain_o2n(a,ttrian)
+end
+
+function CellData.change_domain(a::CellField,strian::AdaptedTriangulation,::ReferenceDomain,ttrian::Triangulation,::ReferenceDomain)
+  if strian === ttrian
+    return a
+  end
+  @assert is_change_possible(strian,ttrian)
+
+  if (get_background_model(strian) === get_background_model(ttrian))
+    return change_domain(a,strian.trian,ReferenceDomain(),ttrian,ReferenceDomain())
+  end
+  
+  return change_domain_n2o(a,ttrian)
+end
+
+function CellData.change_domain(a::CellField,strian::AdaptedTriangulation,::ReferenceDomain,ttrian::AdaptedTriangulation,::ReferenceDomain)
+  if is_child(strian,ttrian) # fine to coarse
+    return change_domain(a,strian,ReferenceDomain(),ttrian.trian,ReferenceDomain())
+  else # coarse to fine
+    return change_domain(a,strian.trian,ReferenceDomain(),ttrian,ReferenceDomain())
+  end
+end
+
+for sdomain in [:ReferenceDomain,:PhysicalDomain]
+  for (stype,ttype) in [(:AdaptedTriangulation,:AdaptedTriangulation),(:AdaptedTriangulation,:Triangulation),(:Triangulation,:AdaptedTriangulation)]
+    @eval begin
+      function CellData.change_domain(a::CellField,strian::$stype,::$sdomain,ttrian::$ttype,::PhysicalDomain)
+        a_ref  = change_domain(a,ReferenceDomain())
+        atrian = change_domain(a_ref,strian,ReferenceDomain(),ttrian,ReferenceDomain())
+        return change_domain(atrian,PhysicalDomain())
+      end
+    end
+  end
+end
+
+function Geometry.move_contributions(scell_to_val::AbstractArray, strian::AdaptedTriangulation, ttrian::Triangulation)
+  smodel = get_adapted_model(strian)
+  @check get_parent(smodel) === get_background_model(ttrian)
+
+  tcell_to_val = move_contributions(scell_to_val,get_adaptivity_glue(smodel))
+  return tcell_to_val
+end
+
+function Geometry.move_contributions(scell_to_val::AbstractArray, glue::AdaptivityGlue)
+  tcell_to_scells = glue.o2n_faces_map
+  k = Geometry.CombineContributionsMap(scell_to_val)
+  tcell_to_val = lazy_map(k,tcell_to_scells)
+  return tcell_to_val
+end
+
+function change_domain_o2n(f_old, new_trian::AdaptedTriangulation{Dc,Dp}) where {Dc,Dp}
+  @notimplementedif num_dims(get_triangulation(f_old)) != Dc
```

Comment:
Here we issue notimplement error if the Dc-dimension does not match. 

Question: Let us assume that `Dc < Dp`. Does the code below also work in that case? If not, shouldnt the condition also reflect that?

Reply:
I would say it does work. I don't believe we are testing it, since (as mentioned in other comments) I am not currently testing any exotic case. However this particular case should work verbatim: Since I am only reordering fields, it does not matter which dimension the `Point`s on which I am evaluating these fields actually live, which is in my head the only thing that changes (i.e we might be in parametric 2 dimensions but points in the physical space might have more dimensions, right?). Does this make sense? 
In any case, I do believe it could be good to test some of these exotic cases.